TriblerGUI/widgets/channelcontentswidget.py

```diff
@@ -0,0 +1,461 @@
+                CATEGORY_SELECTOR_ITEMS.index(self.model.category_filter)
+                if self.model.category_filter in CATEGORY_SELECTOR_ITEMS
+                else 0
+            )
+            self.model.info_changed.connect(self.on_model_info_changed)
+            self.controller.set_model(self.model)
+            self.controller.table_view.resizeEvent(None)
+            self.update_labels()
+
+    def on_channel_clicked(self, channel_dict):
+        self.initialize_with_channel(channel_dict)
+
+    # TODO: restore the method and button to copy channel_id to the clipboard
+    # def on_copy_channel_id(self):
+    #    copy_to_clipboard(self.channel_info["public_key"])
+    #    self.tray_show_message("Copied channel ID", self.channel_info["public_key"])
+
+    def preview_clicked(self):
+        params = {
+            'query_filter': self.model.channel_info["public_key"],
+            'metadata_type': 'torrent',
+            'rest_endpoint_url': 'search',
+            'first': 1,
+            'last': 50,
+        }
+        self.model.perform_query(**params)
+
+    def create_new_channel(self):
+        self.model.create_new_channel()
+
+    def initialize_with_channel(self, channel_info):
+        # Turn off sorting by default to speed up SQL queries
+        self.content_table.horizontalHeader().setSortIndicator(-1, Qt.AscendingOrder)
+        self.push_channels_stack(self.default_channel_model(channel_info=channel_info))
+        self.controller.set_model(self.model)
+        self.controller.table_view.resizeEvent(None)
+
+        self.channel_preview_button.setHidden(channel_info['state'] in ('Complete', 'Legacy', 'Persona'))
+        self.content_table.setFocus()
+        self.details_container.hide()
+        self.channel_torrents_filter_input.setText("")
+        self.channel_options_button.show()
+        self.model.reset()
+
+        self.update_labels()
+
+    def update_labels(self, dirty=False):
+        # initialize the page about a channel
+        self.channel_name_label.setText(self.model.channel_info['name'])
+
+        if self.model.channel_info.get("state", None) == "Personal":
+            self.channel_options_button.setHidden(False)
+        else:
+            self.channel_options_button.setHidden(True)
+
+        if self.model.channel_info.get("state", None) == "Personal":
+            self.new_channel_button.setHidden(False)
+            self.channel_state_label.setHidden(True)
+        else:
+            self.new_channel_button.setHidden(True)
+            self.channel_state_label.setHidden(False)
+            if "state" in self.model.channel_info:
+                self.channel_state_label.setText(self.model.channel_info["state"])
+                self.subscription_widget.update_subscribe_button(self.model.channel_info)
+
+        # self.total_items = response['total']
+        # if self.num_results_label:
+        #    self.num_results_label.setText("%d results" % self.total_items)
```

Comment:
Does not seem to be present in the .ui file (if I'm not wrong) so might as well remove it.

Reply:
Fixed